servant-server/src/Servant/Server/Internal/RoutingApplication.hs

```diff
@@ -119,7 +123,7 @@ runAction action respond k = do
       e <- runEitherT a
       respond $ case e of
         Right x  -> k x
-        Left err -> succeedWith $ responseServantErr err
+        Left err -> failWithServantError err
```

Comment:
I think this is generally the right idea, but it's really unclear to me whether we want to just generally fail without further ado *here*.